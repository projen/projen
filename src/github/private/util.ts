export function secretToString(secretName: string): string {
  return `\${{ secrets.${secretName} }}`;
}

export function context(value: string) {
  return `\${{ ${value} }}`;
}

// Checks if part of the file path is hidden
export function isHiddenPath(path: string) {
  return /(^|\/)\.[^\/\.]/g.test(path);
}

// Helper to assert a path is not hidden
export function ensureNotHiddenPath(value: string, name: string) {
  if (isHiddenPath(value)) {
    throw Error(`${name} cannot be a hidden path, got: ${value}`);
  }
}
