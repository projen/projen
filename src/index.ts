export * from "./ai-instructions";
export * from "./compare";
export * from "./component";
export * from "./dev-env";
export * from "./dependencies";
export * from "./docker-compose";
export * from "./dockerfile";
export * from "./file";
export * from "./gitattributes";
export * from "./gitpod";
export * from "./runner-options";
export * from "./ignore-file";
export * from "./ini";
export * from "./json";
export * from "./json-patch";
export * from "./logger";
export * from "./license";
export * from "./makefile";
export * from "./object-file";
export * from "./option-hints";
export * from "./project";
export * from "./project-build";
export * from "./project-tree";
export * from "./projects";
export * from "./projenrc";
export * from "./projenrc-json";
export * from "./readme";
export * from "./renovatebot";
export * from "./sample-file";
export * from "./semver";
export * from "./source-code";
export * from "./task";
export * from "./tasks";
export * from "./task-model";
export * from "./task-runtime";
export * from "./testing";
export * from "./textfile";
export * from "./toml";
export * from "./version";
export * from "./yaml";
export * from "./xmlfile";

// export submodules
export * as web from "./web";
export * as github from "./github";
export * as gitlab from "./gitlab";
export * as circleci from "./circleci";
export * as vscode from "./vscode";
export * as java from "./java";
export * as python from "./python";
export * as typescript from "./typescript";
export * as javascript from "./javascript";
export * as release from "./release";
export * as awscdk from "./awscdk";
export * as cdk8s from "./cdk8s";
export * as cdktf from "./cdktf";
export * as cdk from "./cdk";
export * as build from "./build";
