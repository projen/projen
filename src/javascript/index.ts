export * from "./bundler";
export * from "./eslint";
export * from "./jest";
export * from "./license-checker";
export * from "./node-package";
export * from "./node-project";
export * from "./npm-config";
export * from "./prettier";
export * from "./projenrc";
export * from "./typescript-config";
export * from "./upgrade-dependencies";
export * from "./yarnrc";
