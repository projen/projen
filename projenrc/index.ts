import { DependencyType, JsonFile, Project, TextFile } from "../src";
import { PROJEN_MARKER } from "../src/common";
import { TaskWorkflow } from "../src/github";
import {
  NodeProject,
  UpgradeDependencies,
  UpgradeDependenciesSchedule,
} from "../src/javascript";

export * from "./windows-build";

/**
 * Add integration tests tasks to a project
 *
 * @param project The project to add the tasks to
 */
export function setupIntegTest(project: NodeProject) {
  const packagePythonTask = project.tasks.tryFind("package:python");

  const pythonCompatTask = project.addTask("integ:python-compat", {
    exec: "scripts/python-compat.sh",
    description:
      "Checks that projen's submodule structure does not cause import failures for python. Expects python to be installed and projen to be fully built.",
  });
  const integTask = project.addTask("integ", {
    description: "Run integration tests",
  });
  integTask.spawn(project.compileTask);
  if (packagePythonTask) {
    integTask.spawn(packagePythonTask);
  }
  integTask.spawn(pythonCompatTask);

  project.buildWorkflow?.addPostBuildJobTask(integTask, {
    tools: { python: { version: "3.x" }, go: { version: "1.16.x" } },
  });
}

/**
 * Add a task to bundle the run-task script needed for "projen eject"
 *
 * @param project The project to add the task to
 */
export function setupBundleTaskRunner(project: Project) {
  // build `run-task` script needed for "projen eject" functionality
  // TODO: use project.bundler.addBundle instead - currently it's too inflexible on where the output goes
  const taskRunnerPath = "lib/run-task.cjs";
  const task = project.addTask("bundle:task-runner", {
    description: 'Bundle the run-task.cjs script needed for "projen eject"',
    exec:
      `esbuild src/task-runtime.ts ` +
      `--outfile=${taskRunnerPath} ` +
      `--bundle ` +
      `--platform=node ` +
      `--format=cjs ` +
      `--external:"*/package.json" ` +
      `--banner:js="#!/usr/bin/env node" ` +
      `--footer:js="const runtime = new TaskRuntime(\\".\\");\nruntime.runTask(process.argv[2]);"`,
  });
  project.postCompileTask.spawn(task);
}

/**
 * Setup the projen bootstrap script
 *
 * @param project The project to add the tasks to
 * @param bootstrapScriptFile The path to the bootstrap script
 */
export function setupProjenBootstrap(
  project: NodeProject,
  bootstrapScriptFile: string
) {
  // this script is what we use as the projen command in this project
  // it will compile the project if needed and then run the cli.
  const bootstrapScript = new TextFile(project, bootstrapScriptFile, {
    executable: true,
    marker: true,
    lines: `#!/usr/bin/env node
// ${PROJEN_MARKER}

const { existsSync } = require("fs");
const { execSync } = require("child_process");

function execCommand(command) {
  try {
    execSync(command, { stdio: "inherit" });
  } catch (error) {
    console.error(\`Failed to execute command: \${command}\`, error);
    process.exit(1);
  }
}

const isBuild = existsSync("lib/cli/index.js");
const hasJsii = existsSync("node_modules/.bin/jsii");
const hasTsNode = existsSync("node_modules/.bin/ts-node");
const needsBootstrapping = !isBuild || !hasTsNode;

const installCommand = "yarn install --frozen-lockfile --check-files --non-interactive";
const buildCommand = "npx jsii --silence-warnings=reserved-word --no-fix-peer-dependencies";

function bootstrap() {
  console.info("bootstrapping...");

  if (!hasTsNode || !hasJsii) {
    execCommand(installCommand);
  }

  if (!isBuild) {
    execCommand(buildCommand);
  }
}

if (needsBootstrapping) {
  bootstrap();
}

const args = process.argv.slice(2).join(" ");
execCommand(\`node bin/projen \${args}\`);
`.split("\n"),
  });

  project.npmignore?.exclude(`/${bootstrapScript.path}`);
}

/**
 * Add a task to check licenses
 *
 * @param project The project to add the task to
 */
export function setupCheckLicenses(project: NodeProject) {
  if (project.github) {
    const installCiTask = project.tasks.tryFind("install:ci");
    const checkLicensesTask = project.tasks.tryFind("check-licenses");

    if (installCiTask && checkLicensesTask) {
      // Run license checker as a separate CI job
      new TaskWorkflow(project.github, {
        name: "check-licenses",
        jobId: "check-licenses",
        triggers: {
          pullRequest: {},
          workflowDispatch: {},
          mergeGroup: {},
          push: { branches: ["main"] },
        },
        permissions: {},
        preBuildSteps: [
          {
            name: "Install dependencies",
            run: project.runTaskCommand(installCiTask),
          },
        ],
        task: checkLicensesTask,
      });
    }
  }
}

/**
 * Add a task to generate API docs
 *
 * @param project The project to add the task to
 */
export function setupJsiiDocgen(project: NodeProject) {
  const docgenTask = project.tasks.tryFind("docgen");

  if (docgenTask) {
    docgenTask.reset(
      "jsii-docgen .jsii -o docs/api/projen --split-by-submodule"
    );
  }

  // ignoring the entire docusaurus folder because it's not needed in the published package
  project.npmignore?.exclude("/docusaurus/");
}

/**
 * Add a task to upgrade dependencies
 *
 * @param project The project to add the task to
 */
export function setupUpgradeDependencies(project: NodeProject) {
  // Upgrade Dependencies in two parts:
  // a) Upgrade bundled dependencies as a releasable fix
  // b) Upgrade devDependencies as a chore
  new UpgradeDependencies(project, {
    taskName: "upgrade-bundled",
    types: [DependencyType.BUNDLED],
    semanticCommit: "fix",
    pullRequestTitle: "upgrade bundled dependencies",
    workflowOptions: {
      labels: ["auto-approve"],
      // Run projen's daily upgrade (and release) acyclic to the schedule that projects are on so they get updates faster
      schedule: UpgradeDependenciesSchedule.expressions(["0 12 * * *"]),
    },
  });
  new UpgradeDependencies(project, {
    taskName: "upgrade",
    exclude: [
      // exclude the bundled deps
      ...project.deps.all
        .filter((d: any) => d.type === DependencyType.BUNDLED)
        .map((d: any) => d.name),
      // constructs version constraint should not be changed
      "constructs",
      // markmac depends on projen, we are excluding it here to avoid a circular update loop
      "markmac",
    ],
    workflowOptions: {
      labels: ["auto-approve"],
    },
  });
}

/**
 * Add only the gitignore inclusions and exclusions rules that are not related to other tools
 *
 * @param project The project to add the rules to
 */
export function setupGitignore(project: Project) {
  project.gitignore.include("/test/inventory/**");
  project.gitignore.exclude("/.idea");
  project.gitignore.exclude("**/.tool-versions");
}

/**
 * Setup markdown readme macros and markdownlint
 *
 * @param project The project to add the rules to
 */
export function setupMarkdown(project: NodeProject) {
  // expand markdown macros in readme
  const macros = project.addTask("readme-macros");
  macros.exec("mv README.md README.md.bak");
  macros.exec("cat README.md.bak | markmac > README.md");
  macros.exec("rm README.md.bak");
  project.postCompileTask.spawn(macros);

  new JsonFile(project, ".markdownlint.json", {
    obj: {
      default: true,
      "commands-show-output": false,
      "line-length": {
        line_length: 200,
      },
    },
  });
  project.npmignore?.exclude("/.markdownlint.json");
}

/**
 * Setup vscode configuration
 *
 * @param project The project to add the configuration to
 */
export function setupVscode(project: NodeProject) {
  project.vscode?.launchConfiguration.addConfiguration({
    type: "pwa-node",
    request: "launch",
    name: "projen CLI",
    skipFiles: ["<node_internals>/**"],
    program: "${workspaceFolder}/lib/cli/index.js",
    outFiles: ["${workspaceFolder}/lib/**/*.js"],
  });
}

/**
 * Setup mergify rules
 * @param project The project to add the rules to
 */
export function setupMergify(project: NodeProject) {
  project.github?.mergify?.addRule({
    name: "Label core contributions",
    actions: {
      label: {
        add: ["contribution/core"],
      },
    },
    conditions: ["author~=^(eladb|Chriscbr)$", "label!=contribution/core"],
  });
}

/**
 * Setup gitpod configuration
 *
 * @param project The project to add the configuration to
 */
export function setupGitpod(project: NodeProject) {
  project.gitpod?.addCustomTask({
    name: "Setup",
    init: "yarn install",
    prebuild: "bash ./projen.bash",
    command: "npx projen build",
  });
}

/**
 * Setup devcontainer configuration
 *
 * @param project The project to add the configuration to
 */
export function setupDevcontainer(project: NodeProject) {
  const setup = project.addTask("devenv:setup");
  setup.exec("yarn install");
  setup.spawn(project.buildTask);
  project.devContainer?.addTasks(setup);

  project.npmignore?.exclude("/.devcontainer.json");
}

/**
 * Setup all-contributors
 *
 * @param project The project to add the configuration to
 */
export function setupAllContributors(project: NodeProject) {
  project.addTask("contributors:update", {
    exec: 'all-contributors check | grep "Missing contributors" -A 1 | tail -n1 | sed -e "s/,//g" | xargs -n1 | grep -v "\\[bot\\]" | grep -v "cdklabs-automation" | xargs -n1 -I{} all-contributors add {} code',
  });
  project.npmignore?.exclude("/.all-contributorsrc");
}

/**
 * Add the npmignore rules that are not related to other tools
 *
 * @param project The project to add the rules to
 */
export function setupNpmignore(project: NodeProject) {
  project.npmignore?.exclude("/docs/");
  project.npmignore?.exclude("/logo/");
  project.npmignore?.exclude("/rfcs/");
  project.npmignore?.exclude("/scripts/");
  project.npmignore?.exclude("/ARCHITECTURE.md");
  project.npmignore?.exclude("/CODE_OF_CONDUCT.md");
  project.npmignore?.exclude("/CONTRIBUTING.md");
  project.npmignore?.exclude("/VISION.md");
  project.npmignore?.exclude("/SECURITY.md");
  project.npmignore?.exclude("/.gitpod.yml");
}

export function setupBiomeTypesGeneration(project: NodeProject) {
  const genetateBiomeTypesTask = project.addTask("generate-biome-types", {
    exec: "node ./scripts/generate-biome-types.js",
  });

  const preCompileTask = project.tasks.tryFind("pre-compile");
  // Add type generation to build step to include it also to self-mutation
  if (preCompileTask) {
    preCompileTask.insertStep(preCompileTask.steps.length, {
      spawn: genetateBiomeTypesTask.name,
    });
  }
}
