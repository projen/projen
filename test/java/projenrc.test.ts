import * as path from "path";
import { ProjectOption } from "../../lib/inventory";
import { generateJavaOptionNames } from "../../lib/java";
import { Pom } from "../../src/java";
import { Projenrc, getJavaImport } from "../../src/java/projenrc";
import { renderProjenInitOptions } from "../../src/javascript/render-options";
import { synthSnapshot, TestProject, withProjectDir } from "../util";

test("projenrc.java support", () => {
  // GIVEN
  const project = new TestProject();
  const pom = new Pom(project, {
    groupId: "my.group.id",
    artifactId: "hello-world",
    version: "1.2.3",
  });

  // WHEN
  new Projenrc(project, pom, {
    projenVersion: "1.0.0",
  });

  // THEN
  expect(synthSnapshot(project)).toMatchSnapshot();
});

test("use a non-test scope", () => {
  // GIVEN
  const project = new TestProject();
  const pom = new Pom(project, {
    groupId: "my.group.id",
    artifactId: "hello-world",
    version: "1.2.3",
  });

  // WHEN
  new Projenrc(project, pom, {
    testScope: false,
    projenVersion: "7.7.7",
  });

  // THEN
  expect(synthSnapshot(project)).toMatchSnapshot();
});

test("set the class name", () => {
  // GIVEN
  const project = new TestProject();
  const pom = new Pom(project, {
    groupId: "my.group.id",
    artifactId: "hello-world",
    version: "1.2.3",
  });

  // WHEN
  new Projenrc(project, pom, {
    className: "boom.bam.projenrc",
    projenVersion: "7.7.7",
  });

  // THEN
  expect(synthSnapshot(project)).toMatchSnapshot();
});

test("generate projenrc in java", () => {
  // GIVEN
  const project = new TestProject(
    renderProjenInitOptions("projen.java.JavaProject", {})
  );
  const pom = new Pom(project, {
    groupId: "my.group.id",
    artifactId: "hello-world",
    version: "1.2.3",
  });

  // WHEN
  new Projenrc(project, pom);

  // THEN
  expect(
    synthSnapshot(project)["src/test/java/projenrc.java"]
  ).toMatchSnapshot();
});

test("assert unknown manifest type doesn't throw", () => {
  // GIVEN
  const options: ProjectOption[] = [
    {
      fqn: "unknown.fqn",
      path: [],
      name: "unknownFqn",
      simpleType: "",
      switch: "",
      fullType: {},
      parent: "",
    },
    {
      fqn: "known.fqn",
      path: [],
      name: "knownFqn",
      simpleType: "",
      switch: "",
      fullType: {},
      parent: "",
    },
  ];

  const jsiiManifest: any = {
    types: {
      "known.fqn": {
        namespace: "known_namespace",
        name: "component",
      },
    },
  };

  // WHEN
  const optionNames = generateJavaOptionNames(options, jsiiManifest);

  // THEN
  const optionNameKeys = Object.keys(optionNames);
  expect(optionNameKeys.length).toEqual(1);
  expect(optionNames[optionNameKeys[0]]).toEqual("known_namespace.component");
});

test("assert getJavaImport returns the correct import for submodules.", () => {
  // GIVEN

  const jsiiTypeWithSubmodule = {
    assembly: "@aws/lib",
    namespace: "sub_module",
    name: "Component",
  };

  const jsiiManifest: any = {
    submodules: {
      "@aws/lib.sub_module": {
        targets: {
          java: {
            package: "software.aws.sdk.submodule",
          },
        },
      },
    },
    types: {
      "@aws/lib.sub_module.Component": jsiiTypeWithSubmodule,
    },
  };

  // WHEN
  const fullNameWithSubmodule = getJavaImport(
    jsiiTypeWithSubmodule,
    jsiiManifest
  );

  // THEN
  expect(fullNameWithSubmodule).toEqual("software.aws.sdk.submodule.Component");
});

test("assert getJavaImport returns the correct import with no submodules", () => {
  // GIVEN
  const jsiiType = {
    assembly: "@aws/lib",
    name: "Component",
  };

  const jsiiManifest: any = {
    types: {
      "@aws/lib.Component": jsiiType,
    },
    targets: {
      java: {
        package: "software.aws.sdk",
      },
    },
  };

  // WHEN
  const fullName = getJavaImport(jsiiType, jsiiManifest);

  // THEN
  expect(fullName).toEqual("software.aws.sdk.Component");
});

test("assert generateProjenrc returns the correct projenrc with correct outdir", () => {
  withProjectDir((projectdir) => {
    const newOutDir = path.join(projectdir, "foo");
    const project = new TestProject(
      renderProjenInitOptions("projen.java.JavaProject", {
        outdir: newOutDir,
      })
    );
    const pom = new Pom(project, {
      groupId: "my.group.id",
      artifactId: "hello-world",
      version: "1.2.3",
    });
    const projen = new Projenrc(project, pom);
    expect(projen.project.outdir).toEqual(newOutDir);
  });
});
