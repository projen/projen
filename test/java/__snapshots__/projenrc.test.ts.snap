// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate projenrc in java 1`] = `
"import io.github.cdklabs.projen.java.JavaProject;
import io.github.cdklabs.projen.java.JavaProjectOptions;

public class projenrc {
    public static void main(String[] args) {
        JavaProject project = new JavaProject();
        project.synth();
    }
}"
`;

exports[`projenrc.java support 1`] = `
{
  ".gitattributes": "# ~~ Generated by projen. To modify, edit src/test/java/projenrc.java and run "npx projen".

* text=auto eol=lf
/.gitattributes linguist-generated
/.github/workflows/pull-request-lint.yml linguist-generated
/.gitignore linguist-generated
/.projen/** linguist-generated
/.projen/deps.json linguist-generated
/.projen/files.json linguist-generated
/.projen/tasks.json linguist-generated
/pom.xml linguist-generated",
  ".github/workflows/pull-request-lint.yml": "# ~~ Generated by projen. To modify, edit src/test/java/projenrc.java and run "npx projen".

name: pull-request-lint
on:
  pull_request_target:
    types:
      - labeled
      - opened
      - synchronize
      - reopened
      - ready_for_review
      - edited
  merge_group: {}
jobs:
  validate:
    name: Validate PR title
    runs-on: ubuntu-latest
    permissions:
      pull-requests: write
    if: (github.event_name == 'pull_request' || github.event_name == 'pull_request_target')
    steps:
      - uses: amannn/action-semantic-pull-request@v6
        env:
          GITHUB_TOKEN: \${{ secrets.GITHUB_TOKEN }}
        with:
          types: |-
            feat
            fix
            chore
          requireScope: false
",
  ".gitignore": "# ~~ Generated by projen. To modify, edit src/test/java/projenrc.java and run "npx projen".
node_modules/
!/.gitattributes
!/.projen/tasks.json
!/.projen/deps.json
!/.projen/files.json
!/.github/workflows/pull-request-lint.yml
!/pom.xml
!/src/test/java/projenrc.java
",
  ".projen/deps.json": {
    "//": "~~ Generated by projen. To modify, edit src/test/java/projenrc.java and run "npx projen".",
    "dependencies": [
      {
        "name": "org.codehaus.mojo/exec-maven-plugin",
        "type": "build",
        "version": "3.0.0",
      },
      {
        "name": "io.github.cdklabs/projen",
        "type": "test",
        "version": "1.0.0",
      },
    ],
  },
  ".projen/files.json": {
    "//": "~~ Generated by projen. To modify, edit src/test/java/projenrc.java and run "npx projen".",
    "files": [
      ".gitattributes",
      ".github/workflows/pull-request-lint.yml",
      ".gitignore",
      ".projen/deps.json",
      ".projen/files.json",
      ".projen/tasks.json",
      "pom.xml",
    ],
  },
  ".projen/tasks.json": {
    "//": "~~ Generated by projen. To modify, edit src/test/java/projenrc.java and run "npx projen".",
    "tasks": {
      "build": {
        "description": "Full release build",
        "name": "build",
        "steps": [
          {
            "spawn": "default",
          },
          {
            "spawn": "pre-compile",
          },
          {
            "spawn": "compile",
          },
          {
            "spawn": "post-compile",
          },
          {
            "spawn": "test",
          },
          {
            "spawn": "package",
          },
        ],
      },
      "compile": {
        "description": "Only compile",
        "name": "compile",
      },
      "default": {
        "description": "Synthesize project files",
        "name": "default",
        "steps": [
          {
            "exec": "mvn compiler:testCompile --quiet",
          },
          {
            "exec": "mvn exec:java --quiet -Dexec.mainClass=projenrc -Dexec.classpathScope="test"",
          },
        ],
      },
      "eject": {
        "description": "Remove projen from the project",
        "env": {
          "PROJEN_EJECTING": "true",
        },
        "name": "eject",
        "steps": [
          {
            "spawn": "default",
          },
        ],
      },
      "package": {
        "description": "Creates the distribution package",
        "name": "package",
      },
      "post-compile": {
        "description": "Runs after successful compilation",
        "name": "post-compile",
      },
      "pre-compile": {
        "description": "Prepare the project for compilation",
        "name": "pre-compile",
      },
      "test": {
        "description": "Run tests",
        "name": "test",
      },
    },
  },
  "README.md": "# replace this",
  "pom.xml": "<?xml version="1.0" encoding="UTF-8"?>
<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
    <modelVersion>4.0.0</modelVersion>
    <groupId>my.group.id</groupId>
    <artifactId>hello-world</artifactId>
    <version>1.2.3</version>
    <packaging>jar</packaging>
    <name>my-project</name>
    <build>
        <plugins>
            <plugin>
                <groupId>org.codehaus.mojo</groupId>
                <artifactId>exec-maven-plugin</artifactId>
                <version>3.0.0</version>
            </plugin>
        </plugins>
    </build>
    <dependencies>
        <dependency>
            <groupId>io.github.cdklabs</groupId>
            <artifactId>projen</artifactId>
            <version>1.0.0</version>
            <scope>test</scope>
        </dependency>
    </dependencies>
</project>
<!-- ~~ Generated by projen. To modify, edit src/test/java/projenrc.java and run "npx projen". -->",
}
`;

exports[`set the class name 1`] = `
{
  ".gitattributes": "# ~~ Generated by projen. To modify, edit src/test/java/boom/projenrc.java and run "npx projen".

* text=auto eol=lf
/.gitattributes linguist-generated
/.github/workflows/pull-request-lint.yml linguist-generated
/.gitignore linguist-generated
/.projen/** linguist-generated
/.projen/deps.json linguist-generated
/.projen/files.json linguist-generated
/.projen/tasks.json linguist-generated
/pom.xml linguist-generated",
  ".github/workflows/pull-request-lint.yml": "# ~~ Generated by projen. To modify, edit src/test/java/boom/projenrc.java and run "npx projen".

name: pull-request-lint
on:
  pull_request_target:
    types:
      - labeled
      - opened
      - synchronize
      - reopened
      - ready_for_review
      - edited
  merge_group: {}
jobs:
  validate:
    name: Validate PR title
    runs-on: ubuntu-latest
    permissions:
      pull-requests: write
    if: (github.event_name == 'pull_request' || github.event_name == 'pull_request_target')
    steps:
      - uses: amannn/action-semantic-pull-request@v6
        env:
          GITHUB_TOKEN: \${{ secrets.GITHUB_TOKEN }}
        with:
          types: |-
            feat
            fix
            chore
          requireScope: false
",
  ".gitignore": "# ~~ Generated by projen. To modify, edit src/test/java/boom/projenrc.java and run "npx projen".
node_modules/
!/.gitattributes
!/.projen/tasks.json
!/.projen/deps.json
!/.projen/files.json
!/.github/workflows/pull-request-lint.yml
!/pom.xml
!/src/test/java/boom/projenrc.java
",
  ".projen/deps.json": {
    "//": "~~ Generated by projen. To modify, edit src/test/java/boom/projenrc.java and run "npx projen".",
    "dependencies": [
      {
        "name": "org.codehaus.mojo/exec-maven-plugin",
        "type": "build",
        "version": "3.0.0",
      },
      {
        "name": "io.github.cdklabs/projen",
        "type": "test",
        "version": "7.7.7",
      },
    ],
  },
  ".projen/files.json": {
    "//": "~~ Generated by projen. To modify, edit src/test/java/boom/projenrc.java and run "npx projen".",
    "files": [
      ".gitattributes",
      ".github/workflows/pull-request-lint.yml",
      ".gitignore",
      ".projen/deps.json",
      ".projen/files.json",
      ".projen/tasks.json",
      "pom.xml",
    ],
  },
  ".projen/tasks.json": {
    "//": "~~ Generated by projen. To modify, edit src/test/java/boom/projenrc.java and run "npx projen".",
    "tasks": {
      "build": {
        "description": "Full release build",
        "name": "build",
        "steps": [
          {
            "spawn": "default",
          },
          {
            "spawn": "pre-compile",
          },
          {
            "spawn": "compile",
          },
          {
            "spawn": "post-compile",
          },
          {
            "spawn": "test",
          },
          {
            "spawn": "package",
          },
        ],
      },
      "compile": {
        "description": "Only compile",
        "name": "compile",
      },
      "default": {
        "description": "Synthesize project files",
        "name": "default",
        "steps": [
          {
            "exec": "mvn compiler:testCompile --quiet",
          },
          {
            "exec": "mvn exec:java --quiet -Dexec.mainClass=boom.bam.projenrc -Dexec.classpathScope="test"",
          },
        ],
      },
      "eject": {
        "description": "Remove projen from the project",
        "env": {
          "PROJEN_EJECTING": "true",
        },
        "name": "eject",
        "steps": [
          {
            "spawn": "default",
          },
        ],
      },
      "package": {
        "description": "Creates the distribution package",
        "name": "package",
      },
      "post-compile": {
        "description": "Runs after successful compilation",
        "name": "post-compile",
      },
      "pre-compile": {
        "description": "Prepare the project for compilation",
        "name": "pre-compile",
      },
      "test": {
        "description": "Run tests",
        "name": "test",
      },
    },
  },
  "README.md": "# replace this",
  "pom.xml": "<?xml version="1.0" encoding="UTF-8"?>
<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
    <modelVersion>4.0.0</modelVersion>
    <groupId>my.group.id</groupId>
    <artifactId>hello-world</artifactId>
    <version>1.2.3</version>
    <packaging>jar</packaging>
    <name>my-project</name>
    <build>
        <plugins>
            <plugin>
                <groupId>org.codehaus.mojo</groupId>
                <artifactId>exec-maven-plugin</artifactId>
                <version>3.0.0</version>
            </plugin>
        </plugins>
    </build>
    <dependencies>
        <dependency>
            <groupId>io.github.cdklabs</groupId>
            <artifactId>projen</artifactId>
            <version>7.7.7</version>
            <scope>test</scope>
        </dependency>
    </dependencies>
</project>
<!-- ~~ Generated by projen. To modify, edit src/test/java/boom/projenrc.java and run "npx projen". -->",
}
`;

exports[`use a non-test scope 1`] = `
{
  ".gitattributes": "# ~~ Generated by projen. To modify, edit src/main/java/projenrc.java and run "npx projen".

* text=auto eol=lf
/.gitattributes linguist-generated
/.github/workflows/pull-request-lint.yml linguist-generated
/.gitignore linguist-generated
/.projen/** linguist-generated
/.projen/deps.json linguist-generated
/.projen/files.json linguist-generated
/.projen/tasks.json linguist-generated
/pom.xml linguist-generated",
  ".github/workflows/pull-request-lint.yml": "# ~~ Generated by projen. To modify, edit src/main/java/projenrc.java and run "npx projen".

name: pull-request-lint
on:
  pull_request_target:
    types:
      - labeled
      - opened
      - synchronize
      - reopened
      - ready_for_review
      - edited
  merge_group: {}
jobs:
  validate:
    name: Validate PR title
    runs-on: ubuntu-latest
    permissions:
      pull-requests: write
    if: (github.event_name == 'pull_request' || github.event_name == 'pull_request_target')
    steps:
      - uses: amannn/action-semantic-pull-request@v6
        env:
          GITHUB_TOKEN: \${{ secrets.GITHUB_TOKEN }}
        with:
          types: |-
            feat
            fix
            chore
          requireScope: false
",
  ".gitignore": "# ~~ Generated by projen. To modify, edit src/main/java/projenrc.java and run "npx projen".
node_modules/
!/.gitattributes
!/.projen/tasks.json
!/.projen/deps.json
!/.projen/files.json
!/.github/workflows/pull-request-lint.yml
!/pom.xml
!/src/main/java/projenrc.java
",
  ".projen/deps.json": {
    "//": "~~ Generated by projen. To modify, edit src/main/java/projenrc.java and run "npx projen".",
    "dependencies": [
      {
        "name": "org.codehaus.mojo/exec-maven-plugin",
        "type": "build",
        "version": "3.0.0",
      },
      {
        "name": "io.github.cdklabs/projen",
        "type": "runtime",
        "version": "7.7.7",
      },
    ],
  },
  ".projen/files.json": {
    "//": "~~ Generated by projen. To modify, edit src/main/java/projenrc.java and run "npx projen".",
    "files": [
      ".gitattributes",
      ".github/workflows/pull-request-lint.yml",
      ".gitignore",
      ".projen/deps.json",
      ".projen/files.json",
      ".projen/tasks.json",
      "pom.xml",
    ],
  },
  ".projen/tasks.json": {
    "//": "~~ Generated by projen. To modify, edit src/main/java/projenrc.java and run "npx projen".",
    "tasks": {
      "build": {
        "description": "Full release build",
        "name": "build",
        "steps": [
          {
            "spawn": "default",
          },
          {
            "spawn": "pre-compile",
          },
          {
            "spawn": "compile",
          },
          {
            "spawn": "post-compile",
          },
          {
            "spawn": "test",
          },
          {
            "spawn": "package",
          },
        ],
      },
      "compile": {
        "description": "Only compile",
        "name": "compile",
      },
      "default": {
        "description": "Synthesize project files",
        "name": "default",
        "steps": [
          {
            "exec": "mvn compiler:compile --quiet",
          },
          {
            "exec": "mvn exec:java --quiet -Dexec.mainClass=projenrc",
          },
        ],
      },
      "eject": {
        "description": "Remove projen from the project",
        "env": {
          "PROJEN_EJECTING": "true",
        },
        "name": "eject",
        "steps": [
          {
            "spawn": "default",
          },
        ],
      },
      "package": {
        "description": "Creates the distribution package",
        "name": "package",
      },
      "post-compile": {
        "description": "Runs after successful compilation",
        "name": "post-compile",
      },
      "pre-compile": {
        "description": "Prepare the project for compilation",
        "name": "pre-compile",
      },
      "test": {
        "description": "Run tests",
        "name": "test",
      },
    },
  },
  "README.md": "# replace this",
  "pom.xml": "<?xml version="1.0" encoding="UTF-8"?>
<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
    <modelVersion>4.0.0</modelVersion>
    <groupId>my.group.id</groupId>
    <artifactId>hello-world</artifactId>
    <version>1.2.3</version>
    <packaging>jar</packaging>
    <name>my-project</name>
    <build>
        <plugins>
            <plugin>
                <groupId>org.codehaus.mojo</groupId>
                <artifactId>exec-maven-plugin</artifactId>
                <version>3.0.0</version>
            </plugin>
        </plugins>
    </build>
    <dependencies>
        <dependency>
            <groupId>io.github.cdklabs</groupId>
            <artifactId>projen</artifactId>
            <version>7.7.7</version>
        </dependency>
    </dependencies>
</project>
<!-- ~~ Generated by projen. To modify, edit src/main/java/projenrc.java and run "npx projen". -->",
}
`;
